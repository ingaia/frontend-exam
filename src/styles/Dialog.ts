import { makeStyles } from '@material-ui/core/styles';

export const useStyles = makeStyles((theme) => ({
  root: {
    backgroundColor: 'var(--black)',
    border: '1px solid var(--hillary)',
    overflow: 'hidden',
    '& iframe': {
      border: 'unset',
      [theme.breakpoints.down('xs')]: {
        height: 'auto'
      },
    }
  }
}));
