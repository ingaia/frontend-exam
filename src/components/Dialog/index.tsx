import React from 'react';
import Dialog from '@material-ui/core/Dialog';
import DialogContent from '@material-ui/core/DialogContent';
import { useStyles } from '../../styles/Dialog';

export interface Props {
  open: boolean;
  handleClose: () => void;
  videoId: string;
};

const Component = ({ open, handleClose, videoId }: Props) => (
  <Dialog onClose={handleClose} maxWidth="md" fullWidth aria-labelledby="simple-dialog-title" open={open}>
    <DialogContent classes={useStyles()}>
      <iframe
        title="Youtube video dialog"
        width="100%"
        height="500"
        src={`https://www.youtube.com/embed/${videoId}`}
        allow="accelerometer; autoplay; clipboard-write; encrypted-media; gyroscope; picture-in-picture"
      />

    </DialogContent>
  </Dialog>
);

export default Component;
