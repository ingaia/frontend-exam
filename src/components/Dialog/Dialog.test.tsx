import React from 'react';
import { render } from '@testing-library/react';
import Dialog from '.';

describe('<Dialog />', () => {
  it('should render correctly', () => {
    const { container, getByTitle } = render(<Dialog open videoId="#test" handleClose={() => {}} />);

    expect(getByTitle('Youtube video dialog').src).toEqual('https://www.youtube.com/embed/#test')
    expect(container.firstChild).toMatchSnapshot()
  })
})
