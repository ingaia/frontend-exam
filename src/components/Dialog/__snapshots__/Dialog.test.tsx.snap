// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Dialog /> should render correctly 1`] = `null`;
